crates/matrix-sdk-indexeddb/src/crypto_store/mod.rs

```diff
@@ -534,7 +537,17 @@ impl IndexeddbCryptoStore {
 
         let mut device_store = indexeddb_changes.get(keys::DEVICES);
 
+        let account_info = self.get_static_account();
+        let mut clear_caches = false;
         for device in device_changes.new.iter().chain(&device_changes.changed) {
+            // If our own device key changes, we need to clear the session
+            // cache because the sessions contain a copy of our device key, and
+            // we want the sessions to use the new version.
+            if account_info.clone().is_some_and(|info| {
+                info.user_id == device.user_id() && info.device_id == device.device_id()
```

Comment:
Should this be `!=`? I don't fully understand the code, but if we're looking for a changed device ID I'd expect to be looking for a different value here.

Reply:
Ah, no - it's a new _key_ for the _same_ device, right?

Reply:
Yup, we only need to clear the cache if we get new device keys for our own device (e.g. if we get new cross-signing signatures).